thousandeyes/resource_ftp_server_test.go

```diff
@@ -0,0 +1,103 @@
+package thousandeyes
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccThousandEyesFTPServer(t *testing.T) {
+	var ftpResourceName = "thousandeyes_ftp_server.test"
+	testCases := []struct {
+		name                 string
+		resourceFile         string
+		resourceName         string
+		checkDestroyFunction func(*terraform.State) error
+		checkFunc            []resource.TestCheckFunc
+	}{
+		{
+			name:                 "basic",
+			resourceFile:         "acceptance_resources/ftp_server/basic.tf",
+			resourceName:         ftpResourceName,
+			checkDestroyFunction: testAccCheckDefaultResourceDestroy,
+			checkFunc: []resource.TestCheckFunc{
+				resource.TestCheckResourceAttr(ftpResourceName, "alerts_enabled", "false"),
+				resource.TestCheckResourceAttr(ftpResourceName, "alert_rules.#", "0"),
+			},
+		},
+		{
+			name:                 "alerts_enabled",
+			resourceFile:         "acceptance_resources/ftp_server/alerts_enabled.tf",
+			resourceName:         ftpResourceName,
+			checkDestroyFunction: testAccCheckDefaultResourceDestroy,
+			checkFunc: []resource.TestCheckFunc{
+				resource.TestCheckResourceAttr(ftpResourceName, "alerts_enabled", "true"),
+				resource.TestCheckResourceAttr(ftpResourceName, "alert_rules.#", "1"),
+			},
+		},
+		{
+			name:         "alerts_enabled_multiple_alert_rules",
+			resourceFile: "acceptance_resources/ftp_server/alerts_enabled_multiple_alert_rules.tf",
+			resourceName: ftpResourceName,
+			checkDestroyFunction: func(state *terraform.State) error {
+				resourceList := []ResourceType{
+					{
+						Name:         "FTP Server Test",
+						ResourceName: "thousandeyes_ftp_server",
+						GetResource: func(id int64) (interface{}, error) {
+							return testClient.GetFTPServer(id)
+						}},
+					{
+						Name:         "Alert Rule",
+						ResourceName: "thousandeyes_alert_rule",
+						GetResource: func(id int64) (interface{}, error) {
+							return testClient.GetAlertRule(id)
+						}},
+				}
+				return testAccCheckResourceDestroy(resourceList, state)
+			},
+			checkFunc: []resource.TestCheckFunc{
+				resource.TestCheckResourceAttr(ftpResourceName, "alerts_enabled", "true"),
```

Comment:
I only scratched the surface here, we can do way more complex validations here:https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep

For bootstrapping the acceptance tests, this is enough since each test tests the following flow:
1- Terraform Plan
2- Terraform Apply
3- New terraform plan to make sure that there are no reported changes
4- Terraform destroy
5- Check, using our `testAccCheckDefaultResourceDestroy` method that the resources were actually destroyed